Modules/gcmodule.c

```diff
@@ -1992,59 +1992,58 @@ static PyMethodDef GcMethods[] = {
     {NULL,      NULL}           /* Sentinel */
 };
 
-static struct PyModuleDef gcmodule = {
-    PyModuleDef_HEAD_INIT,
-    "gc",              /* m_name */
-    gc__doc__,         /* m_doc */
-    -1,                /* m_size */
-    GcMethods,         /* m_methods */
-    NULL,              /* m_reload */
-    NULL,              /* m_traverse */
-    NULL,              /* m_clear */
-    NULL               /* m_free */
-};
-
-PyMODINIT_FUNC
-PyInit_gc(void)
+static int
+gcmodule_exec(PyObject *module)
 {
     GCState *gcstate = get_gc_state();
 
-    PyObject *m = PyModule_Create(&gcmodule);
-
-    if (m == NULL) {
-        return NULL;
-    }
-
+    /* Initialized by _PyGC_Init() early in interpreter lifecycle */
     if (gcstate->garbage == NULL) {
-        gcstate->garbage = PyList_New(0);
-        if (gcstate->garbage == NULL) {
-            return NULL;
-        }
+        PyErr_SetString(PyExc_SystemError,
+                        "GC garbage bin is not initialized");
+        return -1;
     }
-    Py_INCREF(gcstate->garbage);
-    if (PyModule_AddObject(m, "garbage", gcstate->garbage) < 0) {
-        return NULL;
+    if (PyModule_AddObjectRef(module, "garbage", gcstate->garbage) < 0) {
+        return -1;
     }
 
+    assert(gcstate->callbacks == NULL);
+    gcstate->callbacks = PyList_New(0);
```

Comment:
This looks wrong to me. If you create multiple instances of the GC modul, callbacks member will be overriden by a new list at each call. It must be initialized exactly once in _PyGC_Init().

By the way, `if (gcstate->garbage == NULL) {` test in _PyGC_Init() can be removed, it's useless. This function is called exactly once per interpreter.

Reply:
Makes sense, I'm moving ``gcstate->callbacks = PyList_New(0)`` to ``_PyGC_Init()``.